compiler/parser/parser.go

```diff
@@ -603,26 +603,50 @@ func (p *Parser) parseIdent() *ast.Ident {
 	}
 }
 
-func (p *Parser) parseIdentList() *ast.IdentList {
+func (p *Parser) parseIdentList(allowVarArgs bool) *ast.IdentList {
```

Comment:
I guess `allowVarArgs` is always `true` in the current implementation.

Reply:
So this can be reduced to

```golang
func (p *Parser) parseIdentList() *ast.IdentList {
	if p.trace {
		defer un(trace(p, "IdentList"))
	}

	var params []*ast.Ident
	lparen := p.expect(token.LParen)
	isVarArgs := false
	if p.token != token.RParen {
		if p.token == token.Ellipsis {
			isVarArgs = true
			p.next()
		}

		params = append(params, p.parseIdent())
		for !isVarArgs && p.token == token.Comma {
			p.next()
			if p.token == token.Ellipsis {
				isVarArgs = true
				p.next()
			}
			params = append(params, p.parseIdent())
		}
	}

	rparen := p.expect(token.RParen)

	return &ast.IdentList{
		LParen:  lparen,
		RParen:  rparen,
		VarArgs: isVarArgs,
		List:    params,
	}
}
```

Reply:
yeah, i was being proactive but YAGNI, fixed